specs/interop/predeploys.md

```diff
@@ -254,6 +279,62 @@ Note that the `relayMessage` function is `payable` to enable relayers to earn in
 To enable cross chain authorization patterns, both the `_sender` and the `_source` MUST be exposed via `public`
 getters.
 
+#### Sending Back Messages Hashes
+
+When sending back a message that failed to be relayed on the destination chain
+to the source chain, it's crucial to ensure the message can only be sent back
+to the `L2ToL2CrossDomainMessenger` contract in its source chain.
+
+```solidity
+function sendMessageHashBack(uint256 _messageSource, uint256 _nonce, address _sender, address _target, bytes calldata _message) external nonReentrant {
+    if (_source == block.chainid) revert MessageSourceSameChain();
+
+    bytes32 messageHash = keccak256(abi.encode(block.chainid, _messageSource, _nonce, _sender, _target, _message));
+
+    if (successfulMessages[messageHash]) revert MessageAlreadyRelayed();
+    if (block.timestamp <  failedMessages[messageHash] + RETURN_DELAY) revert DelayHasNotEnsued();
```

Comment:
What is the specific purpose of the return delay? Its defined in an invariant but there is no rationale for why it exists. What would happen if it didn't exist? What are the tradeoffs with a larger/smaller value?

Reply:
return delay is there to allow for replay-ability of the message over that time window.
if it wasn't there, there could be a griefing attack where a malicious actor will execute a message with minimum gas, forcing a revert, and then send that message back to the origin domain.
the current time window has been picked at random, we have not done any research on either the best static or dynamic delay for the system.
do you have any other solution in mind besides a time-window? we believe for the MVP this should work ok-ish.